src/main/java/htsjdk/samtools/cram/structure/Slice.java

```diff
@@ -113,6 +119,46 @@ public ReferenceContext getReferenceContext() {
         return referenceContext;
     }
 
+    public void baiIndexInitializationCheck() {
+        final StringBuilder error = new StringBuilder();
+
+        if (byteOffsetFromContainer == UNINITIALIZED_INDEXING_PARAMETER) {
+            error.append("Cannot index this Slice for BAI because its byteOffsetFromContainer is unknown.").append(System.lineSeparator());
+        }
+
+        if (containerByteOffset == UNINITIALIZED_INDEXING_PARAMETER) {
+            error.append("Cannot index this Slice for BAI because its containerByteOffset is unknown.").append(System.lineSeparator());
+        }
+
+        if (index == UNINITIALIZED_INDEXING_PARAMETER) {
+            error.append("Cannot index this Slice for BAI because its index is unknown.").append(System.lineSeparator());
+        }
+
+        if (error.length() > 0) {
+            throw new CRAMException(error.toString());
+        }
+    }
+
+    public void craiIndexInitializationCheck() {
```

Comment:
I'll add a comment to make the difference clearer.

I'd prefer to keep them separate, since they both test for exactly what is needed under their own requirements.

Reply:
I don't see why the distinction, or the increased complexity and code duplication, is useful - there is only one `distributeIndexingParametersToSlices` thats establishes these values. But ok.